tools/python_api/src_py/query_result.py

```diff
@@ -102,14 +112,26 @@ def get_as_pl(self):
         """
         Get the query result as a Polars DataFrame.
 
+        See Also
+        --------
+        get_as_df : Get the query result as a Pandas DataFrame.
+        get_as_arrow : Get the query result as a PyArrow Table.
+
         Returns
         -------
         polars.DataFrame
             Query result as a Polars DataFrame.
         """
 
         import polars as pl
-        return pl.from_arrow(data=self.get_as_arrow(10_000))
+
+        target_n_elems = (
+            10_000_000  # adaptive chunk_size; target 10m elements per chunk
+        )
+        target_chunk_size = max(target_n_elems // len(self.get_column_names()), 10)
```

Comment:
This makes sense - for most cases in this setting, we'd have a single chunk that helps Polars retrieve it more efficiently, and it's also in line with the DuckDB heuristically selected value of 1M.

Reply:
Yup; each chunk will be broadly equivalent to 10 cols of 1m rows each.